src/Dofs/DofHandler.jl

```diff
@@ -104,99 +153,204 @@ function Base.show(io::IO, dh::DofHandler)
         print(io, "  Dofs per cell: ", ndofs_per_cell(dh))
     end
 end
+immutable SortedNTuple{N, T}
+  n::NTuple{N, T}
+  SortedNTuple(n::NTuple{N, T}) = new{N,T}(swapsort(n))
+end
+SortedNTuple{N,T}(n::NTuple{N, T}) = SortedNTuple{N,T}(n)
 
-ndofs(dh::DofHandler) = length(dh.dofs_nodes)
-ndofs_per_cell(dh::DofHandler) = size(dh.dofs_cells, 1)
-isclosed(dh::DofHandler) = dh.closed[]
-dofs_node(dh::DofHandler, i::Int) = dh.dof_nodes[:, i]
-
-# Stores the dofs for the cell with number `i` into the vector `global_dofs`
-function celldofs!(global_dofs::Vector{Int}, dh::DofHandler, i::Int)
-    @assert isclosed(dh)
-    @assert length(global_dofs) == ndofs_per_cell(dh)
-    @inbounds for j in 1:ndofs_per_cell(dh)
-        global_dofs[j] = dh.dofs_cells[j, i]
-    end
-    return global_dofs
+immutable GlobalNode{N}
+    dof_numbers::NTuple{N, Int} # Number of the dofs
 end
 
-# Add a collection of fields
-function Base.push!(dh::DofHandler, names::Vector{Symbol}, dims)
-    @assert length(names) == length(dims)
-    for i in 1:length(names)
-        push!(dh, names[i], dims[i])
-    end
+immutable GlobalEdge{N}
+    sign::Bool                  # "sign" of the edge
```

Comment:
I guess. This is whether the cell that created dofs for this edge had a positive or negative edge.